packages/react-dom/src/events/__tests__/SyntheticKeyboardEvent-test.js

```diff
@@ -451,6 +451,30 @@ describe('SyntheticKeyboardEvent', () => {
         });
       });
     });
+
+    describe('code', () => {
+      it('should throw (in dev) when KeyboardEvent.code is undefined', () => {
+        const node = ReactDOM.render(
+          <input
+            onKeyDown={e => {
+              return e.code;
+            }}
+          />,
+          container,
+        );
+        let evt = new KeyboardEvent('keydown', {
+          code: undefined,
+          bubbles: true,
+          cancelable: true,
+        });
+
+        if (__DEV__ && evt.code === undefined) {
```

Comment:
Didn't know you could use this in tests too 🤔 

Do tests get run twice for this, or does it just always take one of the paths?

Reply:
It seems like it should work, I was going off what I saw in other test files like this [here](https://github.com/facebook/react/blob/master/packages/react-dom/src/__tests__/ReactComponent-test.js#L53) It seems like it just runs once and chooses the path dependent on the env variables. 

But I also was unable to find a way to force the `code` property of the `KeyboardEvent` to be undefined to actually test it as the constructor will set it to `''` automatically if set to undefined like in line 466.

Reply:
@Jessidhia 

> Do tests get run twice for this, or does it just always take one of the paths?

[They're run twice 🙂](https://github.com/facebook/react/blob/679402a66b09e129d06f415cd976d18727cc8590/package.json#L105)